pages/nfts/new.tsx

```diff
@@ -166,42 +206,93 @@ export default function BulkUploadWizard() {
     }
   }, [wallet, connect]);
 
-  const transformFormVals = (values: NFTFormValue[], filePins: UploadedFilePin[]): NFTValue[] => {
-    return values.map((v, i: number) => {
-      const filePin = filePins[i];
-      return {
-        name: v.name,
-        description: v.description,
-        symbol: '',
-        collection: {
-          name: v.collectionName,
-          family: v.collectionFamily,
-        },
-        seller_fee_basis_points: v.seller_fee_basis_points,
-        image: filePin.uri,
-        files: [{ uri: filePin.uri, type: filePin.type }],
-        attributes: v.attributes.reduce((result: Array<NFTAttribute>, a: NFTAttribute) => {
-          if (!isNil(a?.trait_type)) {
-            result.push({ trait_type: a.trait_type, value: a.value });
-          }
-          return result;
-        }, []),
-        properties: v.properties,
-      };
-    });
+  const uploadCoverImage = async (file: File) => {
+    const body = new FormData();
+    body.append(file.name, file, file.name);
+
+    try {
+      const resp = await fetch('/api/ipfs/upload', {
+        method: 'POST',
+        body,
+      });
+      const json = await resp.json();
+      if (json) {
+        return json.files[0];
+      }
+    } catch (e) {
+      console.error('Could not upload file', e);
+      throw new Error(e);
+    }
+  };
+
+  const transformFormVals = async (values: NFTFormValue[], filePins: UploadedFilePin[]) => {
+    const resp = await Promise.all(
+      values.map(async (v, i: number) => {
+        const filePin = filePins[i];
+
+        if (!filePin) {
+          throw new Error('No file pin for index ' + i);
+        }
+
+        const category: NFTCategory = detectCategoryByFileExt(filePin.name);
+        const isMultiMedia = category !== 'image';
+
+        let coverImageFile: UploadedFilePin | undefined;
+        if (v.coverImageFile) {
+          coverImageFile = await uploadCoverImage(v.coverImageFile);
```

Comment:
Moving upload out of this transform would separate the concerns. Ideally transform can happen synchronously.